### DISCUSSION TITLE
Getting a TypeError: '<' not supported between instances of 'float' and 'complex' training SNARIMAX

### DISCUSSION THREAD

**Discussion by RaphaelChevallier** [online-ml/river#947] [Q&A]
Hello,

I am trying to implement a snarimax time series model. I have regular spaced intervals of data. I have exactly 179 dicts points of data every 15 mins. I use a for loop to train the model one dict at a time then forecast the next 15 mins value.
```
for i in data:
        training = model.learn_one(x = i, y = close)
        trainingLog = modelLog.learn_one(x = i, y = y)
    y_pred = model.forecast(horizon=1)
    y_predLog = modelLog.forecast(horizon=1)
```

However during this for loop training model I get some dicts that issue an error on the training method model.learn_one. `TypeError: '<' not supported between instances of 'float' and 'complex'`. This is odd as they are all the same dict values. Every entry has the same key values and the value types are always floats. I have tried to handle the error and it is very random as to what data throws this

Any ideas would be much appreciated

**Comment by MaxHalford** [comment]
Hey there. First of all you can't pass an `int` (your `i`) to the model like that. It expects a `dict` of features. If I were you I would carefully look at our examples, and ideally run them yourself to understand them.

**Reply by RaphaelChevallier** [reply]
The variable i isn't an int. data is a list of dicts. I am passing a dict to x like in the documentation. Specifically the dict is format {"dateTime":xxxxx}

**Reply by MaxHalford** [reply]
Ok my bad. Well then can provide a reproducible piece of code? I can't debug with what you have provided us.
